File: 박동혁/2주차/1874.cpp
#include<iostream>
#include<stack>
#include<vector>

using namespace std;

int arr[100001];
int main()
{
    int n, num, cnt=1;
    cin >> n;
    stack<int> stk;
    vector<char> ans;

    for(int i=0; i<n; i++) {
        cin >> arr[i];
    }
    for(int i=0; i<n; i++) {
        int num = arr[i];
        if(num >= cnt) {
            for( ; cnt<=num; cnt++) {
                stk.push(cnt);
                ans.push_back('+');
            }
            stk.pop();
            ans.push_back('-');
        } else if(num == stk.top()) {
            stk.pop();
            ans.push_back('-');
        } else {
            cout << "NO";
            return 0;
        }
    }
    for(vector<char>::iterator it=ans.begin(); it<ans.end(); it++) {
        cout << *it << '\n';
    }
}

File: 박혜린/2주차/1874.py
n = int(input())
stack, ans = [], []
cur = 1
isPossible = True

for i in range(n):
  num = int(input())
  while cur <= num: # push
    stack.append(cur)
    ans.append('+')
    cur += 1
  if stack[-1] == num: # pop
    stack.pop()
    ans.append('-')
  else:
    isPossible = False

if isPossible :
  for i in ans:
    print(i)
else:
  print('NO')
